File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/LongFieldType.java
package com.example.frameworks.domain.core;

public class LongFieldType extends FieldType<Long> {
    public static FieldType<Long> asLong() {
        return new LongFieldType();
    }

    @Override
    protected boolean match(Class<?> underlyingType) {
        return Number.class.isAssignableFrom(underlyingType);
    }

    @Override
    protected Long convert(Object value) {
        return ((Number) value).longValue();
    }
}

File: frameworks/payload.core/src/main/kotlin/com/qomolangma/frameworks/payload/core/LongFieldType.kt
package com.qomolangma.frameworks.payload.core

class LongFieldType : FieldType<Long>() {
    override fun match(underlyingType: Class<*>): Boolean {
        return Number::class.java.isAssignableFrom(underlyingType)
    }

    override fun convert(value: Any): Long {
        return (value as Number).toLong()
    }

    companion object {
        @JvmStatic
        fun asLong(): FieldType<Long> {
            return LongFieldType()
        }
    }
}
